tests/bal_functional_behaviour_suite.py

```diff
@@ -142,6 +144,54 @@ def test_when_getWithRelationshipsPaged_called_then_results_delayed_by_specified
             self.createTestContext(access=Context.Access.kRead),
         )
 
+    def test_when_getWithRelationshipPaged_pager_hasNext_called_then_results_delayed(
+        self,
+    ):
+        for query_latency in self.__simulated_latency_subtests():
+            pager = self.__getWithRelationshipPaged_pager()
+            with self.__assert_simulated_latency_applied(query_latency):
+                pager.hasNext()
+
+    def test_when_getWithRelationshipsPaged_pager_hasNext_called_then_results_delayed(
+        self,
+    ):
+        for query_latency in self.__simulated_latency_subtests():
+            pager = self.__getWithRelationshipsPaged_pager()
+            with self.__assert_simulated_latency_applied(query_latency):
+                pager.hasNext()
```

Comment:
[Consider] I'd question whether this is the right axis to test, the simulated delay implementation concerns the methods themselves. As this is an internal test of BAL, I think you'd be fine only testing each method via one of the endpoints.

Reply:
Good point, was still in apiComplianceSuite mode when writing these tests.  Will refactor to remove the duplication.

Reply:
De-duped in 6efdaf97 - how's that?